src/server.cpp

```diff
@@ -75,12 +75,23 @@ void Server::setRoot(const std::string& root)
   }
 }
 
+void Server::setAddress(const std::string& addr)
+{
+  if (addr.empty()) return;
+
+  if (addr.find(':') != std::string::npos) { // IPv6
+    m_addr.addr6 = (addr[0] == '[') ? addr.substr(1, addr.length() - 2) : addr; // Remove brackets if any
+  } else {
+    m_addr.addr = addr;
+  }
+}
```

Comment:
One more foolproofness issue with this function - it is written under the (reasonable) assumption that it is called only once. This peculiarity of the implementation should be at least noted in a comment.